Bug 1780387: host-local plugin should be built and executed within container

dougbtv opened the pull request:
Due to issues with the host-local plugin coredumping, host-local should be built
in the sdn build process and executed from within the container instead of on the host.

Originally removed in commit: 6d6d35f2f51c84c86b32723e88661c3cd8e379a0

In reference to BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1780387

@squeed recommended naming the binary to `osdn-host-local` to differentiate it. I also went and put it in a directory named `/opt/cni/container-bin` to differentiate it from the host's `/opt/cni/bin` however, I certainly would like any feedback on that naming.

squeed commented on pkg/network/node/pod.go:
```diff
@@ -33,9 +33,10 @@ import (
-	cniPluginsBinDir = "/host/opt/cni/bin"
+	podInterfaceName               = "eth0"
+	hostLocalDataDir               = "/var/lib/cni/networks"
+	cniPluginsBinDir               = "/host/opt/cni/bin"
```
is cniPluginsBinDir even used anymore? I don't think so.

dougbtv replied:
Good eye!

squeed commented on images/node/Dockerfile.rhel:
```diff
@@ -7,6 +7,7 @@ RUN CGO_ENABLED=0 GO_BUILD_FLAGS="-tags no_openssl" make build GO_BUILD_PACKAGES
 FROM registry.svc.ci.openshift.org/ocp/4.2:base
 COPY --from=builder /go/src/github.com/openshift/sdn/openshift-sdn /usr/bin/
 COPY --from=builder /go/src/github.com/openshift/sdn/sdn-cni-plugin /opt/cni/bin/openshift-sdn
+COPY --from=builder /go/src/github.com/openshift/sdn/host-local /opt/cni/container-bin/osdn-host-local
```
Bikeshed: put this in `/usr/bin/cni/...`

dougbtv replied:
Much cleaner, thanks

squeed commented:
Wow, that was fast :-)
Two minor nits, then let's get this in.

squeed commented:
Something's not quite right:
`Dec 11 00:53:48 ci-op-9ckm4-m-0.c.openshift-gce-devel-ci.internal crio[1912]: 2019-12-11T00:53:48Z [error] error in getting result from AddNetwork: CNI request failed with status 400: 'failed to run IPAM for 2aa39a37f33363f8ae9c0491a30534e8a21188fcb27cf186af5de605da4f8ad6: failed to run CNI IPAM ADD: fork/exec /opt/cni/container-bin/osdn-host-local: no such file or directory`

dougbtv commented:
/retest

squeed commented:
/hold
so that it stops re-testing. Feel free to self-un-hold.

squeed commented:
same error: Dec 11 16:47:28 ip-10-0-141-223 crio[1952]: 2019-12-11T16:47:28Z [error] error in getting result from AddNetwork: CNI request failed with status 400: 'failed to run IPAM for 5c46d6adf80476f0088ad7c58954a7c85d6503cc79e0ee2b6875c5c5a4407e3e: failed to run CNI IPAM ADD: fork/exec /opt/cni/container-bin/osdn-host-local: no such file or directory

dougbtv commented:
(turns out it helps when you actually `git add` your changes! <insert:facepalm.gif>)

dougbtv commented:
@squeed -- I think I've got it. I had only changed one Dockerfile, but, found more -- and changed all those which referenced the openshift-sdn CNI, which I thought might be an indicator that I had all the correct places.

Those wind up being:

```
images/node/Dockerfile.rhel
images/sdn/Dockerfile.fedora
images/sdn/Dockerfile.rhel
```

(Previously I had just modified `images/node/Dockerfile.rhel`)

dougbtv commented:
/test e2e-aws

(looked like a flake to me, seemed to timeout after approx 4 hours. also passes on the release-4.3 backport, fwiw)

squeed commented:
/hold cancel

squeed commented:
/approve
/lgtm

nee1esh commented:
/retest e2e-gcp
